### DISCUSSION TITLE
Explanation of code

### DISCUSSION THREAD

**Discussion by kojak357** [gopro/labs#959] [Q&A] [answered]
Hi guys

I have started Time-lapse with GoPro h12 but is very much a beginner. I have code here and try to work out what the commands are in this code. Can anyone help perhaps please. 

mPdP>07:29<22:00!S+!1N+!119RQ!3600RQ

mP = Mode Picture
dP = Default picture Mode

>07:29<22:00  = from to time

But the bit I do not understand is "S+!1N+!" and the " RQ "
Does the S mean sleep for a certain amount of time ?
Can someone tell me the meaning of the following please:
!
S+
1N+
RQ

Thanks in advance.

**Comment by dnewman-gpsw** [comment, marked as answer]
```>07:29<22:00!S+!1N+!119RQ```


Are all part of one command conditional unit. The > is an if statement (docs here https://gopro.github.io/labs/control/actions/)

```
if(time >= 07:29 AND time < 22:00)   // >07:29<22:00
{
   Start(); // !S 
   Sleep(1);  // pause 1 second, !1N
   RepeatQuick(119);  // Repeat command after sleeping 119 seconds
} 
```

All the pluses are used to add commands to the true condition of the if statement.

**Reply by kojak357** [reply]
Thank you!
